schema_cache.rst

```diff
@@ -0,0 +1,111 @@
+.. _schema_cache:
+
+Schema Cache
+============
+
+There are several PostgREST features that need information from the database schema.
+If they accessed this information directly from the database every time they needed it, it would be too costly.
+That is why, when PostgREST starts, it generates a database schema cache and uses it to get the information needed for these features:
+
+- For :ref:`resource_embedding`, it looks for foreign key constraints to determine relationships between tables.
+- For :ref:`s_procs`, it looks for their metadata: parameters, return type, volatility and `overloading <https://www.postgresql.org/docs/current/xfunc-overload.html>`_.
+- For :ref:`upsert` statements, it looks for the tables' primary keys.
+- For INSERT statements, it also looks for the tables' primary keys in order to return the Location header.
+- For OPTION requests, it verifies the existence of tables and views and looks for INSTEAD OF triggers for the latter.
+- For :ref:`open-api`, it looks for information about tables (columns, primary keys and foreign keys), views (columns and INSTEAD OF triggers) and stored procedures (metadata).
+
+The Stale Schema Cache
+----------------------
+
+When you make a change in the database related to any of the features mentioned above while PostgREST is running, the schema cache turns stale. You then need to reload the schema before you make a request related to these changes, otherwise you'll receive an error instead of the expected response.
+
+For instance, suppose you add a ``cities`` table to your database. This table has a foreign key ``country_id`` referencing an existing ``countries`` table. Then, you make a request to get the data from ``cities`` and their ``countries``:
+
+.. code-block:: http
+
+  GET /cities?select=name,country:countries(id,name) HTTP/1.1
+
+But instead, you get an error message that looks like this:
+
+.. code-block:: json
+
+  {
+    "hint": "If a new foreign key between these entities was created in the database, try reloading the schema cache.",
+    "message": "Could not find a relationship between cities and countries in the schema cache"
+  }
+
+As you can see, PostgREST couldn't find the relationship in the schema cache.
+
+You get a similar result for stored procedures. Suppose you create this function on the database while PostgREST is running:
```

Comment:
```suggestion

Stale function metadata
~~~~~~~~~~~~~~~~~~

You get a similar result for stored procedures. Suppose you create this function on the database while PostgREST is running:
```

Reply:
Also, let's change "procedures" for "functions" everywhere in this page. We actually don't support pg 11 stored procedures([ref](https://www.postgresql.org/docs/11/sql-createprocedure.html)), so this could cause confusion.

Reply:
Yes, fully agree with that one!

I think we also need to improve https://postgrest.org/en/latest/api.html#stored-procedures.

We should make it explicitly clear over there, that we don't support procedures, but functions - even when we name the concept a "remote procedure call - RPC". Not sure if we should use the term "stored procedures" at all, though.

Reply:
> Also, let's change "procedures" for "functions" everywhere in this page. We actually don't support pg 11 stored procedures(ref), so this could cause confusion.

Oh, I get it. But does this mean that the reference about [stored procedures](https://postgrest.org/en/v7.0.0/api.html#stored-procedures) should be renamed in the future?

Reply:
> Oh, I get it. But does this mean that the reference about stored procedures should be renamed in the future?

Yeah, there's an issue related to that: https://github.com/PostgREST/postgrest-docs/issues/147. For now, we can use "function" instead of "procedure" on the schema cache page.